### PR TITLE
Implement string parsing for the KQL parser.

### PR DESCRIPTION
**Pull request by afoucret**
This PR implement text extraction for the KQL parser quoted and unquoted strings.

### CODE REVIEW

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+ * Provides methods for type-safe parsing, text extraction, and string escaping/unescaping.
+ */
+public final class ParserUtils {
+
+    private static final String UNQUOTED_LITERAL_TERM_DELIMITER = " ";
+    private static final char ESCAPE_CHAR = '\\';
+    private static final char QUOTE_CHAR = '"';
+
+    private ParserUtils() {
+
```
Comment (carlosdelest) [inline x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java:34]: Icing on the non-instantiable cake: 
```suggestion
     throw new UnsupportedOperationException("No need to instantiate this class");
```

**Review by tteofili — APPROVED** [review]
LGTM thanks Aurelien ;)

**Review by carlosdelest — APPROVED** [review]
LGTM! 💯

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    private ParserUtils() {
+        throw new UnsupportedOperationException("No need to instantiate this class");
+    }
+
+    /**
+     * Performs type-safe parsing using the provided visitor.
+     *
+     * @param visitor The visitor to use to do the parsing
+     * @param ctx The parser tree to visit
```
Comment (kderusso) [inline x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java:39]:
```suggestion
     * @param ctx The parser tree context to visit
```

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+     *
+     * @param queryText The query text to escape
+     * @param preserveWildcards If true, does not escape wildcard characters (*)
+     * @return The escaped query string
+     */
+    public static String escapeLuceneQueryString(String queryText, boolean preserveWildcards) {
+        if (preserveWildcards) {
+            StringBuilder escapedQuery = new StringBuilder(queryText.length());
+            StringBuilder subpart = new StringBuilder(queryText.length());
+            for (int i = 0; i < queryText.length(); i++) {
```
Comment (kderusso) [inline x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java:105]: I think you could write this a little more readable, something like: 

```
for (char currentChar : queryText.toCharArray()) {
    if (currentChar == '*') {
        escapedQuery.append(QueryParser.escape(subpart.toString())).append(currentChar);
        subpart.setLength(0); 
    } else {
        subpart.append(currentChar);
    }
}
```

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+                char currentChar = queryText.charAt(i);
+                if (currentChar == '*') {
```
Comment (kderusso) [inline x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java:107]: Nitpick - would be nice to use a constant for `WILDCARD_CHAR`

File: x-pack/plugin/kql/src/test/java/org/elasticsearch/xpack/kql/parser/ParserUtilsTests.java
Comment (kderusso) [inline x-pack/plugin/kql/src/test/java/org/elasticsearch/xpack/kql/parser/ParserUtilsTests.java:1]: Nice tests!

**Review by kderusso — APPROVED** [review]
Nice work!

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java (hunk 3 above)
Comment (afoucret) [reply x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java]: Done. Thank you!

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java (hunk 4 above)
Comment (afoucret) [reply x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java]: Done. Thank you!

File: x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+/**
+ * Utility class for parsing and processing KQL expressions.
+ * Provides methods for type-safe parsing, text extraction, and string escaping/unescaping.
+ */
+public final class ParserUtils {
+
+    private static final String UNQUOTED_LITERAL_TERM_DELIMITER = " ";
+    private static final char ESCAPE_CHAR = '\\';
+    private static final char QUOTE_CHAR = '"';
+    private static final char WILDCARD_CHAR = '"';
```
Comment (afoucret) [inline x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParserUtils.java:30]:
```suggestion
    private static final char WILDCARD_CHAR = '*';

**Pull request merged**